### DISCUSSION TITLE
Trying to fientune for brazilian portuguese language

### DISCUSSION THREAD

**Discussion by abhisirka2001** [SWivid/F5-TTS#184] [Q&A]
I'm fine-tuning the F5 base model for Brazilian Portuguese using 160 hours of audio data with custom tokenizer. After 100 epochs on a single RTX 4090, I've encountered the following issues:

When using an English audio prompt and reference text, the output captures the accent but doesn't produce the expected words. You can listen to the output [here](https://drive.google.com/file/d/1f-Juge1AT7F9tqZYirAhoFVPGru1JfND/view?usp=sharing).

When using a Brazilian Portuguese audio prompt and reference text, the model generates an empty audio file.

What should I consider or adjust to improve the model's performance from here? Any advice on resolving these issues would be appreciated!

**Comment by LucianoDaluz** [comment]
Please let us know if you get it.

**Reply by jeanfredson** [reply]
[SilentAntagonist/F5ttsptbr](https://huggingface.co/SilentAntagonist/F5ttsptbr/tree/main)

**Reply by LucianoDaluz** [reply]
404 page :(

**Reply by dougrhis** [reply]
404 page

**Comment by grebsu** [comment]
eu espero que você consiga!!!!

**Reply by abhisirka2001** [reply]
Hey,
I have tried more on this and now i am getting this output for brazilian portuguese.
can you help me know whether it sounds brazilian or i m going in wrong direction?

https://drive.google.com/file/d/1K3yKhSpMXLZkJer9WKGc6hC1ZZu7BUeN/view?usp=sharing

Transcript: O Brasil, oficialmente República Federativa do Brasil, é o maior país da América do Sul e da América Latina. É o quinto maior país do mundo em área territorial e o sexto em população, com mais de 211 milhões de habitantes.

Thanks

**Reply by gmag11** [reply]
Sounds like some words are missing

**Reply by abhisirka2001** [reply]
Yes can you rate it out of 10 ? how is the accent?

**Reply by LRhaunter** [reply]
As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent.

**Reply by Reza2kn** [reply]
Would love to hear more about your set up here. the dataset size in hours, configs, steps, etc. 🤗🔥

**Reply by abhisirka2001** [reply]
> Would love to hear more about your set up here. the dataset size in hours, configs, steps, etc. 🤗🔥

sure i will share it here soon here

**Reply by abhisirka2001** [reply]
> Would love to hear more about your set up here. the dataset size in hours, configs, steps, etc. 🤗🔥

Hey
I trained the model on more datasets and here is one sample. please let me know if words and accent is better now.
I hope model is good in learning new accent and new language.
https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing

Thanks

**Reply by abhisirka2001** [reply]
> As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent.

Hey
I trained the model on more datasets and here is one sample. please let me know if words and accent is better now.
I hope model is good in learning new accent and new language.
https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing

Thanks

**Reply by LucianoDaluz** [reply]
It is 100% better. It just seems to insert some commas (some pauses) where it shouldnt. But now we can perfectly understand the words.
The only word it says wrong is "Contry = País", that there was a slight lack of intonation in the letter "i"

**Reply by LRhaunter** [reply]
> > As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent.
> 
> Hey I trained the model on more datasets and here is one sample. please let me know if words and accent is better now. I hope model is good in learning new accent and new language. https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing
> 
> Thanks

Now it’s much better! The accent sounds really good, but it still needs a few tweaks. There was a small error in one of the words where a letter was missing.

**Reply by LucianoDaluz** [reply]
> > As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent.
> 
> Hey I trained the model on more datasets and here is one sample. please let me know if words and accent is better now. I hope model is good in learning new accent and new language. https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing
> 
> Thanks

Hey friend! Any update on this one? If you have any new file to check just let me know. All the best.

**Reply by dougrhis** [reply]
@abhisirka2001 You can shared this model?

**Reply by Falkker** [reply]
> > > As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent.
> > 
> > 
> > Hey I trained the model on more datasets and here is one sample. please let me know if words and accent is better now. I hope model is good in learning new accent and new language. https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing
> > Thanks
> 
> Hey friend! Any update on this one? If you have any new file to check just let me know. All the best.

Conseguiu treinar em pt-br amigo?

**Comment by x4080** [comment]
What are the steps to train another language and how much ram needed ?

**Reply by abhisirka2001** [reply]
essential things are preparation of dataset and vocab file for your language and adjusting the model hyperparameters.
i am not sure about the the minimum vram requirement. you can use the gradio finetuning instructions given in the repo.

**Reply by x4080** [reply]
@abhisirka2001 thanks

**Comment by abhisirka2001** [comment]
https://huggingface.co/ModelsLab/F5-tts-brazilian/tree/main

**Reply by traderpedroso** [reply]
> https://huggingface.co/ModelsLab/F5-tts-brazilian/tree/main
> […](#)
> On Tue, Nov 19, 2024, 5:44 AM Falkker ***@***.***> wrote: As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent. Hey I trained the model on more datasets and here is one sample. please let me know if words and accent is better now. I hope model is good in learning new accent and new language. https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing Thanks Hey friend! Any update on this one? If you have any new file to check just let me know. All the best. Conseguiu treinar em pt-br amigo? — Reply to this email directly, view it on GitHub <[#184 (reply in thread)](https://github.com/SWivid/F5-TTS/discussions/184#discussioncomment-11298344)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVBPZ3HPUQZJGK5BTN6NL2D2BJ7FRAVCNFSM6AAAAABQIMKNBSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRZHAZTINA> . You are receiving this because you were mentioned.Message ID: ***@***.***>

I used this as a starting point and have been training for 2 days with over 300 hours, but I still haven't achieved a good result. Besides that, the voice cloning is still terrible. I’m going to add another 1k hours for training on 8 H100s, and we'll see if it can clone with quality. If I manage to make it work, I will share the model, but if it only works with samples that contain private data, sharing won’t be possible. However, I hope it works. Personally, I’ve done fine-tuning on about 3 models, but I wasn't satisfied with any of them. I will share the best samples I have, but remember that you cannot use the samples without the owner's permission, so they are for demonstration purposes only. Some reference audio samples generate with quality and few errors, while the cloning is still terrible. Keep in mind that when generating audio, the reference must always be in the same language as the target output. Even so, up to this point, it fails to produce correct pronunciation for unseen speakers.



[LONG AUDIO GEN](https://drive.google.com/file/d/11xYlftAeR7kSpylFEM14HWhtdqZHnosV/view?usp=sharing)

[SHORT AUDIO GEN](https://drive.google.com/file/d/1dsojVk0GiLDzXmgA5CrZluD0dCns7lEW/view?usp=sharing)

I will leave some crucial points for generating quality audio. First, always use samples that are less than 10 seconds long; otherwise, it will add the excess and create a mix with the generated text and the reference text. Second, always ensure that the reference text matches the audio sample precisely, as Whisper sometimes transcribes incorrectly, so be attentive. Third, to ensure the best quality, use NFE Step 64.

**Reply by sendwebpush** [reply]
Olá @traderpedroso , tbm estou trabalhando para a criação de um modelo, porem esta muito longe de ser perfeito, estou tentando treinar em um MacBook onde que tem 38c de GPU porem esta muito lento, tentei otimizar o código para aceitar os GPU do Macbook porem não prestou de nada, tem alguma sugestão de como acelerar o treinamento? não sei se poder, mas poderia me compartilhar seu contato para falarmos sobre o assunto?

**Reply by LucianoDaluz** [reply]
> > https://huggingface.co/ModelsLab/F5-tts-brazilian/tree/main
> > […](#)
> > On Tue, Nov 19, 2024, 5:44 AM Falkker _**@**_._**> wrote: As a Brazilian, I’d say the voice sounds very real and natural, but it’s still a bit confusing. Even as a native speaker, it can be a little hard to understand. Some words are accurate but lack the accent. Hey I trained the model on more datasets and here is one sample. please let me know if words and accent is better now. I hope model is good in learning new accent and new language. https://drive.google.com/file/d/1Adz4EXNa0yC-ZBs_-V3u_ndb-Zv582FV/view?usp=sharing Thanks Hey friend! Any update on this one? If you have any new file to check just let me know. All the best. Conseguiu treinar em pt-br amigo? — Reply to this email directly, view it on GitHub <[#184 (reply in thread)](https://github.com/SWivid/F5-TTS/discussions/184#discussioncomment-11298344)>, or unsubscribe https://github.com/notifications/unsubscribe-auth/AVBPZ3HPUQZJGK5BTN6NL2D2BJ7FRAVCNFSM6AAAAABQIMKNBSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRZHAZTINA . You are receiving this because you were mentioned.Message ID: _**@**_.**_>
> 
> I used this as a starting point and have been training for 2 days with over 300 hours, but I still haven't achieved a good result. Besides that, the voice cloning is still terrible. I’m going to add another 1k hours for training on 8 H100s, and we'll see if it can clone with quality. If I manage to make it work, I will share the model, but if it only works with samples that contain private data, sharing won’t be possible. However, I hope it works. Personally, I’ve done fine-tuning on about 3 models, but I wasn't satisfied with any of them. I will share the best samples I have, but remember that you cannot use the samples without the owner's permission, so they are for demonstration purposes only. Some reference audio samples generate with quality and few errors, while the cloning is still terrible. Keep in mind that when generating audio, the reference must always be in the same language as the target output. Even so, up to this point, it fails to produce correct pronunciation for unseen speakers.
> 
> [LONG AUDIO GEN](https://drive.google.com/file/d/11xYlftAeR7kSpylFEM14HWhtdqZHnosV/view?usp=sharing)
> 
> [SHORT AUDIO GEN](https://drive.google.com/file/d/1dsojVk0GiLDzXmgA5CrZluD0dCns7lEW/view?usp=sharing)
> 
> I will leave some crucial points for generating quality audio. First, always use samples that are less than 10 seconds long; otherwise, it will add the excess and create a mix with the generated text and the reference text. Second, always ensure that the reference text matches the audio sample precisely, as Whisper sometimes transcribes incorrectly, so be attentive. Third, to ensure the best quality, use NFE Step 64.

Although it looks like "Dilma (ex-brazilian president) speaking" it is way better.

**Reply by traderpedroso** [reply]
> Olá@traderpedroso, tbm estou trabalhando para a criação de um modelo, porem está muito longe de ser perfeito, estou treinando em um MacBook onde que tem 38c de GPU porem está muito lento, tentei aprimorar o código para aceitar os GPU do Macbook porem não prestou de nada, tem alguma sugestão de como acelerar o treinamento? não sei se pode, mas poderia me compartilhar seu contato para falarmos sobre o assunto?

tem a versão coreml em onnix

**Reply by krc1983** [reply]
Oi, @abhisirka2001 eu estou tendo resultados ruins com o seu modelo, a voz esta ok, mas a pronuncia esta bastante confusa, acredito que seja pela falta do arquivo vocab.txt de portugues, vc pode me dizer onde conseguir? obg!

**Reply by Roge7Rodrigues** [reply]
Olá @traderpedroso e @sendwebpush, vocês conseguiram algum resultado satisfatório? Eu tenho uma 3060ti, não consigo fazer testes :/

**Reply by krc1983** [reply]
> Olá @traderpedroso e @sendwebpush, vocês conseguiram algum resultado satisfatório? Eu tenho uma 3060ti, não consigo fazer testes :/

Eu tambem tenho uma 3060ti e consigo usar tranquilamente, o modelo nao gera audios muito coerentes iguais os de outras linguas por ai, acretido que ainda vai precisar de mais treinamento...

**Comment by abhisirka2001** [comment]
yes because i didnt train a separate tokenizer for the brazilian portuguese
language and kept the original vocab file of F5TTS because it container
majority of the tokens. You can further finetune my model with a new vocab
with less data to get good results.

**Reply by krc1983** [reply]
If only I knew how... lol but thx anyway.
